weave/weave_client.py

```diff
@@ -90,6 +90,11 @@ def get_ref(obj: Any) -> Optional[ObjectRef]:
     return getattr(obj, "ref", None)
 
 
+def remove_ref(obj: Any) -> None:
+    if get_ref(obj) is not None:
```

Comment:
This is duplicative w/ L671.  Seems like we want all of L671-675 in its own function `remove_stale_ref(obj)` or just inline the `del obj.ref` instead of calling this function

Reply:
Okay I understand, but I think I disagree. Because we want to return early if there is no stale ref, we probably want to keep that logic in the `_save_nested_objects` fn rather than a helper for the 4 lines. We absolutely could do the `del` inline, but its sort of ugly. If there was any meaningful computational hit to doing that attr check (`get_ref`) then sure, but it should be darn near instant, and we get a safer helper function (`remove_ref`) that mirrors the `get_ref` impl. 

Does that make sense? @nickpenaranda

Reply:
I think remove ref is just setting `obj.ref = None` -- should be ok to inline

Reply:
ah right, I misunderstood what we're returning from.  ok.

re: parity w/ `get_ref`, we may not necessarily want that, as one mutates the object and the other does not.  what happens to the old ref once removed? i guess this is already an issue anyway.